messaging/src/main/java/org/axonframework/eventhandling/deadletter/jpa/JpaSequencedDeadLetterQueue.java

```diff
@@ -0,0 +1,772 @@
+ * {@link DeadLetterEntry#getProcessingStarted()} property, locking other processes out of the sequence for the
+ * configured {@code claimDuration} (30 seconds by default).
+ * <p>
+ * The stored {@link DeadLetterEntry entries} are converted to a {@link JpaDeadLetter} when they need to be processed or
+ * filtered. In order to restore the original {@link EventMessage} a matching {@link DeadLetterJpaConverter} is used.
+ * The default supports all {@code EventMessage} implementations provided by the framework. If you have a custom
+ * variant, you have to build your own.
+ * <p>
+ * {@link org.axonframework.serialization.upcasting.Upcaster upcasters} are not supported by this implementation, so
+ * breaking changes for events messages stored in the queue should be avoided.
+ *
+ * @param <M> An implementation of {@link Message} contained in the {@link DeadLetter dead-letters} within this queue.
+ * @author Mitchell Herrijgers
+ * @since 4.6.0
+ */
+public class JpaSequencedDeadLetterQueue<M extends EventMessage<?>> implements SequencedDeadLetterQueue<M> {
+
+    private static final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+    private final String processingGroup;
+    private final EntityManagerProvider entityManagerProvider;
+    private final List<DeadLetterJpaConverter<EventMessage<?>>> converters;
+    private final TransactionManager transactionManager;
+    private final int maxSequences;
+    private final int maxSequenceSize;
+    private final int queryPageSize;
+    private final Serializer serializer;
+    private final Duration claimDuration;
+
+    protected <T extends EventMessage<?>> JpaSequencedDeadLetterQueue(Builder<T> builder) {
+        builder.validate();
+        this.processingGroup = builder.processingGroup;
+        this.maxSequences = builder.maxSequences;
+        this.maxSequenceSize = builder.maxSequenceSize;
+        this.entityManagerProvider = builder.entityManagerProvider;
+        this.transactionManager = builder.transactionManager;
+        this.serializer = builder.serializer;
+        this.converters = builder.converters;
+        this.claimDuration = builder.claimDuration;
+        this.queryPageSize = builder.queryPageSize;
+    }
+
+    /**
+     * Creates a new builder, capable of building a {@link JpaSequencedDeadLetterQueue} according to the provided
+     * configuration. Note that the {@link Builder#processingGroup(String)}, {@link Builder#transactionManager},
+     * {@link Builder#serializer(Serializer)} and {@link Builder#entityManagerProvider} are mandatory for the queue
+     * to be constructed.
+     *
+     * @param <M> The message type
+     * @return The builder
+     */
+    public static <M extends EventMessage<?>> Builder<M> builder() {
+        return new Builder<>();
+    }
+
+    @Override
+    public void enqueue(@Nonnull Object sequenceIdentifier, @Nonnull DeadLetter<? extends M> letter)
```

Comment:
Add javadoc, likely just
`   /**
     * {@inheritDoc}
     */`
Also for other  overrides in this class

Reply:
This is not necessary as it has no javadoc on its own